invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -448,28 +452,36 @@ private Invoice processDryRun_UPCOMING_INVOICE_FILTERING_Invoice(final UUID acco
         return null;
     }
 
-    private Invoice processDryRun_TARGET_DATE_Invoice(final UUID accountId, final LocalDate targetDate, final List<LocalDate> allCandidateTargetDates, final BillingEventSet billingEvents, final List<Invoice> existingInvoices, final InternalCallContext context) throws InvoiceApiException {
+    private Invoice processDryRun_TARGET_DATE_Invoice(final UUID accountId, final LocalDate targetDate, final Set<LocalDate> allCandidateTargetDates, final BillingEventSet billingEvents, final List<Invoice> existingInvoices, final InternalCallContext context) throws InvoiceApiException {
 
-        LocalDate prevLocalDate = null;
-        for (final LocalDate cur : allCandidateTargetDates) {
-            if (cur.compareTo(targetDate) < 0) {
-                prevLocalDate = cur;
-            } else {
+
+        // Since allCandidateTargetDates is a SortedSet, the same comparator will be reused for the priorityQueue
```

Comment:
Is there a specific reason to use a `PriorityQueue` over a `TreeSet` directly (the implementation of `getUpcomingInvoiceCandidateDates` already returns a `TreeSet`, we could change the returned type)?

I'm afraid that if we refactor the `getUpcomingInvoiceCandidateDates` code in the future, `allCandidateTargetDates` could become a generic `Collection` and this assumption would be gone (also, the magic is technically jdk specific).

Reply:
After re-reading code, simply removed the comment. There is no need for a special Comparator as the `PriorityQueue` (and the `TreeSet`) will both use the *natural order of the element* : `LocalDate` implements `Comparable` already.